test/Microsoft.AspNetCore.ResponseCaching.Tests/TestUtils.cs

```diff
@@ -212,14 +213,17 @@ internal static ResponseCachingContext CreateTestContext(ITestSink testSink)
             };
         }
 
-        internal static void AssertLoggedMessages(List<WriteContext> messages, params LoggedMessage[] expectedMessages)
+        internal static void AssertLoggedMessages(IEnumerable<WriteContext> messages, params LoggedMessage[] expectedMessages)
         {
-            Assert.Equal(messages.Count, expectedMessages.Length);
-            for (var i = 0; i < messages.Count; i++)
+            Assert.Equal(messages.Count(), expectedMessages.Length);
+
+            messages.Zip(expectedMessages, (message, expectedMessage) =>
```

Comment:
Ew, just ToList it inside this method.

Reply:
Or, can we use `Assert.Collection` here?

Reply:
Then ToList it, let's not make it too fancy.

Reply:
But LINQ is so awesome.